v03_pipeline/lib/misc/lookup.py

```diff
@@ -0,0 +1,234 @@
+import hail as hl
+
+from v03_pipeline.lib.model import DatasetType
+
+
+def compute_callset_lookup_ht(
+    dataset_type: DatasetType,
+    mt: hl.MatrixTable,
+    project_guid: str,
+) -> hl.Table:
+    sample_id_to_family_guid = hl.dict(
+        {
+            s: family_guid
+            for family_guid, sample_ids in hl.eval(mt.family_samples).items()
+            for s in sample_ids
+        },
+    )
+    ht = mt.select_rows(
+        project_stats=hl.array(
+            [
+                hl.agg.collect(
+                    hl.Struct(
+                        family_guid=sample_id_to_family_guid[mt.s],
+                        **mt.entry,
+                    ),
+                )
+                .group_by(lambda fs: fs.family_guid)
+                .values()
+                .map(
+                    lambda fs: hl.Struct(
+                        family_guid=fs[0].family_guid,
+                        **{
+                            field_name: hl.len(fs.filter(filter_fn))
+                            for field_name, filter_fn in dataset_type.lookup_table_fields_and_genotype_filter_fns.items()
+                        },
+                    ),
+                ),
+            ],
+        ),
+    ).rows()
+    return globalize_ids(ht, project_guid)
+
+
+def globalize_ids(ht: hl.Table, project_guid: str) -> hl.Table:
+    row = ht.take(1)
+    has_project_stats = row and len(row[0].project_stats) > 0
```

Comment:
very minor, but its confusing that `row` is an array of length one. I would either change the name to `rows` or change the definition to `row = ht.take(1)[0]`

Reply:
`ht.take(1)[0]` will `IndexError` in the empty table case.  `row =  ht.take(1)[0] if ht.count() else None` is maybe clearer though?